src/qibolab/_core/instruments/qm/program/instructions.py

```diff
@@ -73,7 +73,7 @@ def _play(
         _play_single_waveform(op, element, parameters, acquisition)
 
     if parameters.phase is not None:
-        qua.reset_frame(element)
```

Comment:
If the phase was already accounted for in the waveform, why were you resetting the frame?

Reply:
Maybe, if a frame reset was needed, it could have always been needed, even when the `parameters.phase is None`. Since `None` should be a relative zero, I guess (in `Pulse.relative_phase` we actually use `0.0` as default, and never `None` - not sure where the `None` is generated).
If instead is not needed, because you're somehow accumulating the relative phases, then it may be never needed.

I'm trying to understand the whole flow, so, I'm still at the level of collecting hints to reconstruct the full picture...

Reply:
> If the phase was already accounted for in the waveform, why were you resetting the frame?

Because we were still using `frame_rotation_2pi` when sweeping phases, and in that case we may need to `reset_frame`. For example doing the following with sweeper:
```py
for theta in sweeper.values:
    RX(0, theta=theta)
    RX(0, theta=0)
```
there will be a frame rotation to apply the first gate and we need to reset the frame before applying the second.

> Maybe, if a frame reset was needed, it could have always been needed, even when the `parameters.phase is None`. 

That's true, I think you can add `reset_frame` in the program even without doing any rotation and it should have no effect. However, I think `reset_frame`, if not used properly, will screw up `VirtualZ` "pulses", that's why I completely removed it from this PR. For example in
```py
RZ(0, theta=np.pi / 4)
RX(0, theta=np.pi / 2)
RX(0, theta=0)
```
the first `np.pi/4` is a virtual-Z and should be maintained throughout the circuit, so if the second `RX` causes a `reset_frame` in the driver, it won't be correct.